libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java

```diff
@@ -50,20 +55,119 @@
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Supplier;
 
 import static java.util.Collections.emptyMap;
 import static java.util.Collections.singletonMap;
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasLength;
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.in;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.nullValue;
 import static org.junit.internal.matchers.ThrowableMessageMatcher.hasMessage;
 
 public class XContentParserTests extends OpenSearchTestCase {
+    private static final Map<XContentType, Supplier<String>> GENERATORS = Map.of(
+        XContentType.JSON,
+        () -> randomAlphaOfLengthBetween(1, JsonXContent.DEFAULT_MAX_STRING_LEN),
+        XContentType.CBOR,
+        () -> randomAlphaOfLengthBetween(1, CborXContent.DEFAULT_MAX_STRING_LEN),
+        XContentType.SMILE,
+        () -> randomAlphaOfLengthBetween(1, SmileXContent.DEFAULT_MAX_STRING_LEN),
+        /* YAML parser limitation */
+        XContentType.YAML,
+        () -> randomRealisticUnicodeOfCodepointLengthBetween(1, 3140000)
+    );
+
+    private static final Map<XContentType, Supplier<String>> OFF_LIMIT_GENERATORS = Map.of(
+        XContentType.JSON,
+        () -> randomAlphaOfLength(JsonXContent.DEFAULT_MAX_STRING_LEN + 1),
+        XContentType.CBOR,
+        () -> randomAlphaOfLength(CborXContent.DEFAULT_MAX_STRING_LEN + 1),
+        XContentType.SMILE,
+        () -> randomAlphaOfLength(SmileXContent.DEFAULT_MAX_STRING_LEN + 1),
+        /* YAML parser limitation */
+        XContentType.YAML,
+        () -> randomRealisticUnicodeOfCodepointLength(3145730)
```

Comment:
Where does the "3145730" value come from?

Reply:
That's from YAML parser error :)

Reply:
https://github.com/snakeyaml/snakeyaml/blob/master/src/main/java/org/yaml/snakeyaml/LoaderOptions.java#L39